### ISSUE TITLE
Connecting "QAction.triggered" to a slot fails (PyQt5, inside QGIS)

### ISSUE THREAD

**Issue by s-m-e** [CabbageDevelopment/qasync#38]
I am trying to connect to a slot about as follows:

```python
class Foo:
    def __init__(self, mainwindow):
        a = QAction("a", mainwindow)
        a.triggered.connect(self.bar)
    @asyncSlot()
    async def bar():
        pass
```

Doing this with a `QAction` object without the involvement of QGIS works just fine - I have tested it by adding a toolbar to your `fetch` example.

Inside QGIS, I get the following traceback:

```
Traceback (most recent call last):
              File "/home/user/.local/share/QGIS/QGIS3/profiles/conda/python/plugins/foo/bar.py", line 123, in _foo
              self.a.triggered.connect(self.bar)
             TypeError: connect() failed between triggered(bool) and bar()
```

"Outside", QGIS also writes to `stderr`:

```
Warning: QObject::connect: Cannot connect QAction::triggered(bool) to (null)::_disconnect()
```

Any idea what is going on here and/or how to debug this?

I played around with `qasync` and `asyncio.sleep` - `qasync` appears to work inside of QGIS in principle. I am setting up `qasync` based on the advice I got in #33 (thanks once more).

---

I am running QGIS 3.18 on Linux, installed from conda-forge, witch includes PyQt5:

- qgis                      3.18.2           py38hdad52e3_1    conda-forge
- pyqt                      5.12.3           py38h578d9bd_7    conda-forge
- pyqt-impl                 5.12.3           py38h7400c14_7    conda-forge
- pyqt5-sip                 4.19.18          py38h709712a_7    conda-forge
- qasync                    0.15.0             pyhd8ed1ab_0    conda-forge

**Comment by s-m-e** [comment]
Ping @ickby

**Comment by ickby** [comment]
from a very brief look it seems triggered is a signal with bool parameter, hence your slot also needs to take a bool parameter. something like 
@asyncSlot(bool)
    async def bar(triggered):
        pass

**Comment by s-m-e** [comment]
@ickby Yeah, I had the same idea. I tried it. I am getting the same result in QGIS, i.e. the same traceback and output on stderr. Ironically, adding the parameters is not necessary outside of QGIS. That's why I added a toolbar to the `fetch` example from this repository for testing with/without `triggered`.

Ugly workaround:

```python

class Foo:
    def __init__(self, mainwindow):
        a = QAction("a", mainwindow)
        a.triggered.connect(self._sync(self.bar))
    async def bar():
        pass
    def _sync(self, func: Callable, *args: Any, **kwargs: Any) -> Callable:
        def wrapper():
            asyncio.create_task(func(*args, **kwargs))
        return wrapper
```

**Comment by tt2468** [comment]
Have you tried subclassing `Foo()` with `QObject`? I was getting the same issue and realized that the slot was `(null)` because the destination was not a QObject.

**Comment by hosaka** [comment]
As correctly pointed out by @tt2468, the class should inherit from `QObject`. Signals/Slots mechanism only works on valid QObjects. In addition, if your signal has a parameter, your `qasyncSlot` should handle that parameter, I just think it's a good practice (but not required as of #71).
